zio-http/shared/src/main/scala/zio/http/Route.scala

```diff
@@ -131,6 +133,57 @@ sealed trait Route[-Env, +Err] { self =>
         Handled(rpm.routePattern, handler2, location)
     }
 
+  /**
+   * Effectfully peeks at the unhandled failure of this Route.
+   */
+  final def tapErrorZIO[Err1 >: Err](
+    f: Err => ZIO[Any, Err1, Any],
+  )(implicit trace: Trace, ev: CheckResponse[Err]): Route[Env, Err1] =
+    self match {
+      case Provided(route, env)                        => Provided(route.tapErrorZIO(f), env)
+      case Augmented(route, aspect)                    => Augmented(route.tapErrorZIO(f), aspect)
+      case Handled(routePattern, handler, location)    =>
+        Handled(
+          routePattern,
+          if (ev.isResponse) {
+            handler.map(_.tapErrorCauseZIO {
+              case err: Fail[_] =>
+                f(err.value.asInstanceOf[Err]).catchAllCause(cause => ZIO.fail(Response.fromCause(cause)))
+              case _            =>
+                ZIO.unit
+            })
+          } else handler,
+          location,
+        )
+      case Unhandled(rpm, handler, zippable, location) => Unhandled(rpm, handler.tapErrorZIO(f), zippable, location)
+    }
+
+  /**
+   * Effectfully peeks at the unhandled failure cause of this Route.
+   */
+  final def tapErrorCauseZIO[Err1 >: Err](
+    f: Cause[Err] => ZIO[Any, Err1, Any],
+  )(implicit trace: Trace, ev: CheckResponse[Err]): Route[Env, Err1] =
+    self match {
+      case Provided(route, env)                        => Provided(route.tapErrorCauseZIO(f), env)
+      case Augmented(route, aspect)                    => Augmented(route.tapErrorCauseZIO(f), aspect)
+      case Handled(routePattern, handler, location)    =>
+        Handled(
+          routePattern,
+          handler.map(_.tapErrorCauseZIO {
+            case cause0 if ev.isResponse =>
```

Comment:
`ev.isResponse` is always true. Also the concrete type of `f` here is `Cause[Nothing] => ZIO[Any, Err1, Any]`

Reply:
Having `f` as `Cause[Nothing] => ZIO[Any, Err1, Any]` doesn't seem to work as it makes the call to `handler.tapErrorCauseZIO` in the `Unhandled`-case sad.